adagucserverEC/CCairoPlotter.cpp

```diff
@@ -1044,3 +1078,129 @@ void CCairoPlotter::writeToWebP32Stream(FILE *fp, unsigned char, int quality) {
 }
 
 #endif
+
+void CCairoPlotter::drawBarb(int x, int y, double direction, double strength, CColor barbColor, CColor outlineColor, bool drawOutline, float lineWidth, bool toKnots, bool flip) {
+
+  // Preserve path
+  cairo_save(cr);
+  cairo_path_t *cp = cairo_copy_path(cr);
+  cairo_new_path(cr);
+
+  // Draw
+  double wx1, wy1, wx2, wy2, dx1, dy1;
+  int strengthInKnots = round(strength);
+  if (toKnots) {
+    strengthInKnots = round(strength * 3600 / 1852.);
+  }
+
+  if (strengthInKnots <= 2) {
+    // Low wind, only draw a circle
+    cairo_arc(cr, x, y, 6, 0, 2 * M_PI);
+
+  } else {
+
+    int shaftLength = 30;
+
+    int nPennants = strengthInKnots / 50;
+    int nBarbs = (strengthInKnots % 50) / 10;
+    int rest = (strengthInKnots % 50) % 10;
+    int nhalfBarbs;
+    if (rest <= 2) {
+      nhalfBarbs = 0;
+    } else if (rest <= 7) {
+      nhalfBarbs = 1;
+    } else {
+      nhalfBarbs = 0;
+      nBarbs++;
```

Comment:
This is where you can get 5 bars (see my comment on the barb plots).  Same code is in CDrawImage.cpp

Reply:
Fixed